platform-sdk/docs/proposals/TSS-Library/TSS-Library.md

```diff
@@ -0,0 +1,852 @@
+
+###### `TssShareSignature`
+**Description**: Represents a signature created from a TSSPrivateShare.
+
+**Link**:[TssShareSignature.java](tss%2FTssShareSignature.java)
+
+##### Examples:
+###### 1. Bootstrapping
+```java
+   TssService service = new TssService(signatureScheme, new Random());
+   PairingPrivateKey persistentParticipantKey = Requirements.loadECPrivateKey();
+   List<PairingPublicKey> persistentPublicKeys = Requirements.loadECPublicKeys();
+   TssParticipantDirectory participantDirectory =
+           TssParticipantDirectory.createBuilder()
+           .self(/*Identification, different for each participant*/ 0, persistentParticipantKey)
+           .withParticipant(/*Identification:*/0, /*Number of Shares*/5, persistentPublicKeys.get(0))
+           .withParticipant(/*Identification:*/1, /*Number of Shares*/2, persistentPublicKeys.get(1))
+           .withParticipant(/*Identification:*/2, /*Number of Shares*/1, persistentPublicKeys.get(2))
+           .withParticipant(/*Identification:*/3, /*Number of Shares*/1, persistentPublicKeys.get(3))
+           .withThreshold(5)     
+          .build(signatureScheme);
+   
+   //One can then query the directory
+   int n = participantDirectory.getTotalNumberOfShares();
+   List<TssShareId> privateShares = directory.getOwnedSharesIds();
+   List<TssShareId> shareIds = directory.getShareIds();
+```
+###### 1. Create TssMessage
+```java
+   //Creates a TssMessage out of a randomly generated share 
+   TssMessage message = service.generateTssMessage(directory);
+```
+
+###### 2. Validation of TssMessage
+```java
+static {
+    List<TssMessage> messages = Requirements.receiveTssMessages();
+    for(TssMessage m : messages){
+        service.validate(participantDirectory, m);
+    }
+}
+```
+###### 3. Processing of TssMessage
+```java
+    Set<TssMessage> agreedValidMessages = /*Some previously agreed upon same set of valid messages for all participants*/
+    //Get Private Shares    
+    List<TssPrivateShare> privateShares = service.decryptPrivateShares(participantDirectory, agreedValidMessages);
+    //Get Public Shares
+    List<TssPrivateShare> publicShares = service.computePublicShare(participantDirectory, agreedValidMessages);
+
+```
+######  Sign
+```java
+   byte[] message = Requirements.messageToSign();
+   List<TssShareSignature> signatures = service.sign(privateShares, message);
+
+```
+
+######  Aggregate Signatures
+
+```java
+static {
+    List<TssShareSignature> signatures = Requirements.loadSignatures();
+
+    //Validation of individual signatures
+    List<TssShareSignature> validSignatures = new ArrayList<>(signatures.size());
+    for (TssShareSignature signature : signatures) {
+        if (service.verify(participantDirectory, publicShares, signature)) {
+            validSignatures.add(signature);
+        }
+    }
+
+    //Producing an aggregate signature
+    PairingSignature aggregateSignature = service.aggregate(participantDirectory, validSignatures);
```

Comment:
Immaterial change:

I'm not sure if this is the public or private aggregation. Just pointing out it doesn't match the java API